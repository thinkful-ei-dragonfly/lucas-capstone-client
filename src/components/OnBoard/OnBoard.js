import React from 'react'

export default class OnBoard extends React.Component {
  render() {
    return (
      <div className='onboarding-page'>
        <header className='onboarding-header'>
          <h1 className='onboarding-heading-text'>Welcome to Colecci贸n</h1>
        </header>
        <div className='onboarding-content'>
          <h2>Colecci贸n is an interactive, multimedia, collage-style blog leveraging ReactJs, Express, jQuery, Node, PostgreSQL, and SASS</h2>
          <p>When a user is authenticated, they can build Text, Image, and Video posts.</p>
          <p>The author is able to rearrange and resize the posts in a collage-style. Those posts and style changes are stored in PostgreSQL database tables.</p>
          <p>When an unauthenticated end-user visits the page, they are able to interact with the page and manipulate</p>
          <h3>Using Colecci贸n</h3>
          <p>A Demo user has been created (username: <strong>demo_admin</strong>, password: <strong>Demouser123!</strong>)</p>
          <p><em>Additionally, you can register a new user by clicking the 'REGISTER' link</em></p>
          <p>Authenticated users are able to create a new post object. The post creation form will dynamically load relevant field forms based on your selection of post 'type'. You can create Photo, Video, and Text post objects</p>
          <p>Two asynchronous calls are made to the database. The post object is sent to the Colecci贸n Posts table including the title, caption, and other content. Another request is sent to the Styles table, sending the posts new ID as the foreign key.</p>
          <p>As an authenticated user (aka "Author") you can rearrange and resize the posts on your canvas. Using the browser API's <a href='https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver' target='_blank'>MutationObserver</a> interface to listen to DOM inline-style changes, we send PATCH requests to the Styles table to automatically save changes.</p>
          <p>This allows the "author" to design the collage and user-experience to their liking. End users will still be able to interact with the blog and move and resize components to their liking, but only the "author" can make permanent style changes. When the end user refreshes, they see the original styles made by the author</p>
        </div>
      </div>
    )
  }
}
