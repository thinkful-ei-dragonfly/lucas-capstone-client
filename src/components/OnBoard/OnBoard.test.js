import React from 'react'
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json'

import OnBoard from './OnBoard'
configure({ adapter: new Adapter() });
describe('OnBoard component', () => {

  it('renders an OnBoard by default', () => {
    const wrapper = shallow(<OnBoard />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('renders a Header', () => {
    const onboarding_header = shallow(<OnBoard />)
      .find('.onboarding-header')
    expect(toJson(onboarding_header)).toMatchSnapshot()
  })
  it('renders an h1 in the header', () => {
    const onboarding_heading_text = shallow(<OnBoard />)
      .find('.onboarding-heading-text')
    expect(toJson(onboarding_heading_text)).toMatchSnapshot()
  })
  it('renders the onboarding-content section', () => {
    const onboarding_content = shallow(<OnBoard />)
      .find('.onboarding-content')
    expect(toJson(onboarding_content)).toMatchSnapshot()
  })
})
