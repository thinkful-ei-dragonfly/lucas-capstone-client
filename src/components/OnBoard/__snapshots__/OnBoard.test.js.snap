// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnBoard component renders a Header 1`] = `
<header
  className="onboarding-header"
>
  <h1
    className="onboarding-heading-text"
  >
    Welcome to Colección
  </h1>
</header>
`;

exports[`OnBoard component renders an OnBoard by default 1`] = `
<div
  className="onboarding-page"
  role="main"
>
  <header
    className="onboarding-header"
  >
    <h1
      className="onboarding-heading-text"
    >
      Welcome to Colección
    </h1>
  </header>
  <div
    className="onboarding-content"
    role="contentinfo"
  >
    <h2>
      Colección is an interactive, multimedia, collage-style blog leveraging ReactJs, Express, jQuery, Node, PostgreSQL, and SASS
    </h2>
    <p>
      When a user is authenticated, they can build Text, Image, and Video posts.
    </p>
    <p>
      The author is able to rearrange and resize the posts in a collage-style. Those posts and style changes are stored in PostgreSQL database tables.
    </p>
    <p>
      When an unauthenticated end-user visits the page, they are able to interact with the page and manipulate
    </p>
    <h3>
      Using Colección
    </h3>
    <p>
      A Demo user has been created (username: 
      <strong>
        demo_admin
      </strong>
      , password: 
      <strong>
        Demouser123!
      </strong>
      )
    </p>
    <p>
      <em>
        Additionally, you can register a new user by clicking the 'REGISTER' link
      </em>
    </p>
    <p>
      Authenticated users are able to create a new post object. The post creation form will dynamically load relevant field forms based on your selection of post 'type'. You can create Photo, Video, and Text post objects
    </p>
    <p>
      Two asynchronous calls are made to the database. The post object is sent to the Colección Posts table including the title, caption, and other content. Another request is sent to the Styles table, sending the posts new ID as the foreign key.
    </p>
    <p>
      As an authenticated user (aka "Author") you can rearrange and resize the posts on your canvas. Using the browser API's 
      <a
        href="https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver"
        rel="noopener noreferrer"
        target="_blank"
      >
        MutationObserver
      </a>
       interface to listen to DOM inline-style changes, we send PATCH requests to the Styles table to automatically save changes.
    </p>
    <p>
      This allows the "author" to design the collage and user-experience to their liking. End users will still be able to interact with the blog and move and resize components to their liking, but only the "author" can make permanent style changes. When the end user refreshes, they see the original styles made by the author
    </p>
  </div>
</div>
`;

exports[`OnBoard component renders an h1 in the header 1`] = `
<h1
  className="onboarding-heading-text"
>
  Welcome to Colección
</h1>
`;

exports[`OnBoard component renders the onboarding-content section 1`] = `
<div
  className="onboarding-content"
  role="contentinfo"
>
  <h2>
    Colección is an interactive, multimedia, collage-style blog leveraging ReactJs, Express, jQuery, Node, PostgreSQL, and SASS
  </h2>
  <p>
    When a user is authenticated, they can build Text, Image, and Video posts.
  </p>
  <p>
    The author is able to rearrange and resize the posts in a collage-style. Those posts and style changes are stored in PostgreSQL database tables.
  </p>
  <p>
    When an unauthenticated end-user visits the page, they are able to interact with the page and manipulate
  </p>
  <h3>
    Using Colección
  </h3>
  <p>
    A Demo user has been created (username: 
    <strong>
      demo_admin
    </strong>
    , password: 
    <strong>
      Demouser123!
    </strong>
    )
  </p>
  <p>
    <em>
      Additionally, you can register a new user by clicking the 'REGISTER' link
    </em>
  </p>
  <p>
    Authenticated users are able to create a new post object. The post creation form will dynamically load relevant field forms based on your selection of post 'type'. You can create Photo, Video, and Text post objects
  </p>
  <p>
    Two asynchronous calls are made to the database. The post object is sent to the Colección Posts table including the title, caption, and other content. Another request is sent to the Styles table, sending the posts new ID as the foreign key.
  </p>
  <p>
    As an authenticated user (aka "Author") you can rearrange and resize the posts on your canvas. Using the browser API's 
    <a
      href="https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver"
      rel="noopener noreferrer"
      target="_blank"
    >
      MutationObserver
    </a>
     interface to listen to DOM inline-style changes, we send PATCH requests to the Styles table to automatically save changes.
  </p>
  <p>
    This allows the "author" to design the collage and user-experience to their liking. End users will still be able to interact with the blog and move and resize components to their liking, but only the "author" can make permanent style changes. When the end user refreshes, they see the original styles made by the author
  </p>
</div>
`;
